### PR TITLE
Enable dynamic shared memory support for RISC-V

### PR DESCRIPTION
**Pull request by lyctw**
Enable dynamic shared memory support for RISC-V platforms by sharing mobj_dyn_shm.c with ARM platforms and enabling CFG_CORE_DYN_SHM for QEMU virt.

### CODE REVIEW

**Review by jforissier — COMMENTED** [review]
- For "core: mm: move mobj_dyn_shm.c to mm":
  `Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`
- For "riscv: plat-virt: Switch to using dynamic shared memory":
  `Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

For the other commit please see below. Thanks!

File: core/arch/riscv/kernel/boot.c [hunk 1]
```diff
@@ -56,6 +57,168 @@ static void update_external_dt(void)
...
+
+	memcpy(mem, phys_ddr_overall_begin, sizeof(*mem) * nelems);
+	core_mmu_set_discovered_nsec_ddr(mem, nelems);
+}
+#else /*CFG_CORE_DYN_SHM*/
+static void discover_nsec_memory(void)
+{
+}
+#endif /*!CFG_CORE_DYN_SHM*/
+
```
[the remark this answers is not in the document]
Comment (lyctw) [reply core/arch/riscv/kernel/boot.c:221]: Sounds good, thanks for the suggestion.

**Comment by lyctw** [comment]
New commit:
- core: riscv: Add a reserved-memory node for static shared memory

**Comment by jforissier** [comment]
For "core: riscv: Add a reserved-memory node for static shared memory":

`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by lyctw** [comment]
Gentle ping! Hi @jforissier, is there anything to fix or is it ready to merge?

**Comment by jforissier** [comment]
@lyctw all good, thanks for the reminder!

**Pull request merged**